js/models/message_handler.js

```diff
@@ -155,6 +155,27 @@ export default class MessageHandler {
     this.aladin.select(selectionType);
   }
 
+  handleTriggerSelectionByTable(msg) {
+    let sources = msg["table"];
+    let keys = msg["keys"];
+    let sourcesByCatalog = this.aladin.view.catalogs
+      .map((cat) => {
+        if (!cat.isShowing) {
+          return;
+        }
+        return cat.getSources().filter((source) =>
+          sources.some((s) =>
```

Comment:
I found the name `sources` confusing here since they are very different from the `getSources()` and the `source` being filtered.  Perhaps rename to indicate that they are the sources to be selected.  This could tie in with a succinct comment or two here that clarify how the nested functions are filtering the lists.

I'd also be tempted to store the `filter()` result in a local variable and return that on a separate line.  This could help the readability a little and would make any interactive debugging easier.